src/tables.jl

```diff
@@ -47,3 +47,352 @@ function schema_names(featuredefn::IFeatureDefnView)
+
+Convert a table column type to a "clean" type:
+- Unions are flattened
+- Missing and Nothing are dropped
+- Resulting mixed types are approximated by their tightest common supertype
+
+"""
+function _convert_coltype_to_cleantype(T::Type)
+    flattened_T = Base.uniontypes(T)
+    clean_flattened_T = filter(t -> t ∉ [Missing, Nothing], flattened_T)
+    return promote_type(clean_flattened_T...)
+end
+
+function _create_empty_layer_from_AGtypes(
+    strnames::NTuple{N,String},
+    AGtypes::Vector{
+        Union{OGRwkbGeometryType,Tuple{OGRFieldType,OGRFieldSubType}},
+    },
+    name::String,
+) where {N}
+    # Split names and types: between geometry type columns and field type columns
+    geomindices = isa.(AGtypes, OGRwkbGeometryType)
+    !any(geomindices) && error("No column convertible to geometry")
+    geomtypes = AGtypes[geomindices] # TODO consider to use a view
+    geomnames = strnames[geomindices]
+
+    fieldindices = isa.(AGtypes, Tuple{OGRFieldType,OGRFieldSubType})
+    fieldtypes = AGtypes[fieldindices] # TODO consider to use a view
+    fieldnames = strnames[fieldindices]
+
+    # Create layer
+    layer = createlayer(name = name, geom = first(geomtypes))
+    # TODO: create setname! for IGeomFieldDefnView. Probably needs first to fix issue #215
+    # TODO: "Model and handle relationships between GDAL objects systematically"
+    GDAL.ogr_gfld_setname(
+        getgeomdefn(layerdefn(layer), 0).ptr,
+        first(geomnames),
+    )
+
+    # Create FeatureDefn
+    if length(geomtypes) >= 2
+        for (geomtype, geomname) in zip(geomtypes[2:end], geomnames[2:end])
+            creategeomdefn(geomname, geomtype) do geomfielddefn
+                return addgeomdefn!(layer, geomfielddefn) # TODO check if necessary/interesting to set approx=true
+            end
+        end
+    end
+    for (fieldname, (fieldtype, fieldsubtype)) in zip(fieldnames, fieldtypes)
+        createfielddefn(fieldname, fieldtype) do fielddefn
+            setsubtype!(fielddefn, fieldsubtype)
+            return addfielddefn!(layer, fielddefn)
+        end
+    end
+
+    return layer, geomindices, fieldindices
+end
+
+"""
+    _fromtable(sch, rows; name)
+
+Converts a row table `rows` with schema `sch` to a layer (optionally named `name`) within a MEMORY dataset
+
+"""
+function _fromtable end
+
+"""
+    _fromtable(sch::Tables.Schema{names,types}, rows; name::String = "")
+
+Handles the case where names and types in `sch` are different from `nothing`
+
+# Implementation
+1. convert `rows`'s column types given in `sch` to either geometry types or field types and subtypes
+2. split `rows`'s columns into geometry typed columns and field typed columns
+3. create layer named `name` in a MEMORY dataset geomfields and fields types inferred from `rows`'s column types
+4. populate layer with `rows` values
+
+"""
+function _fromtable(
+    sch::Tables.Schema{names,types},
+    rows;
+    name::String,
+    parseWKT::Bool,
+    parseWKB::Bool,
```

Comment:
> > 3. Part of the benefit of the fieldtypes arg is also the ability to skip the try-catch block for inferring the field type:
> 
> The try-catch block should not be time consuming, since it only work on the schema. But I can add a `fieldtypes` as `Union{Nothing, Dict{Int, GeometryOrFieldType}, Dict{String, GeometryOrFieldType}}`

I have added the `fieldtypes` kwarg option, without any specific tests yet in commit 125a076

Reply:
I have added tests on `geomcols` kwarg and corrected some issues raised by those tests in commit 9e90dc14595c1236dd4b4ead44e91ab2b07625a8
-> 10 pts increase on PR coverage.

Reply:
GDAL.jl seems to be broken (failing to precompile) after an update of geos from 3.9 to 3.10

Reply:
I openned an issue in GDAL.jl: https://github.com/JuliaGeo/GDAL.jl/issues/126

Reply:
I have added tests on the `fieldtypes` kwarg. 

Pkg.test is 100% OK and Coverage is 100% when tested locally. So when JuliaGeo/GDAL.jl#126 is fixed, CI for this PR should be OK

@visr or @yeesian do you know how to test error messages as it can be done with other messages by mixing `@test` with `@test_logs`. 
I have not figured out how to mix `@test_throws` and `@test_logs`

Reply:
> @visr or @yeesian do you know how to test error messages as it can be done with other messages by mixing @test with @test_logs.
> I have not figured out how to mix @test_throws and @test_logs

I found a simple way in https://discourse.julialang.org/t/how-to-test-a-default-error-raising-methods-message/19224
`@test_throws ErrorException("message") ex`

Reply:
@yeesian it should be ready for review now